### PR TITLE
Yo boilerplate generator

### PR DESCRIPTION
**Pull request by tomalec**
### Changes proposed in this Pull Request:

Initial PR that moves boilerplate generator from https://github.com/tomalec/generator-ventures/

### Detailed test instructions:

Follow the [generator's README](https://github.com/woocommerce/grow/blob/535ee11f9ca045049478b9a1aa32555e1c93eca8/packages/js/generator-grow/README.md)
1.  Install npm
2. Install Yeoman globally `npm i -g yo`
3. Install this generator globally
	Before the package is published you need to clone and link it manually:
		- `cd packages/js/generator-grow/ && npm link`
4. Go to the folder where you'd like to add boilerplate run `yo grow`

### Changelog entry

> Add - Yeoman generator
> Add - Repository Templates

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Composer
+/vendor/
+
+# Files allowed to be overridden with a local version
+phpcs.xml
+phpunit.xml
+
+# Do not include backup PHP files
+*.php.bak
```
Comment (puntope) [inline .gitignore]: 💅 
At this point we have only `node_modules` maybe also good to add `vendor` in advance, but I would remove all the unnecessary files avoiding noise.

Comment (puntope) [reply .gitignore]: I mean in the grow repo, not the template for the generator.

File: packages/js/generator-grow/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "generator-grow",
+  "version": "0.0.1",
+  "description": "Generate the boilerplate for a WooCommerce plugin repo by Grow Team.",
+  "main": "generators/app/index.js",
+  "type": "module",
+  "scripts": {
+    "test": "mocha generators/**/*.test.js"
```
Comment (puntope) [inline packages/js/generator-grow/package.json:8]: Hello mocha.

**Review by puntope — COMMENTED** [review]
👍  Works like a charm

Left a comment regarding the `.gitingore` for the repo.

**Comment by tomalec** [comment]
Fixed `.gitignore` is there anything more?

**Review by puntope — APPROVED** [review]

**Comment by puntope** [comment]
> Fixed `.gitignore` is there anything more?

Since this PR includes the Readme please merge it before Storybook and I can update the Readme after to add the link to Storybook tool

**Pull request merged**